Tests/ApolloCodegenTests/AnimalKingdomAPI/AnimalKingdomIRCreationTests.swift

```diff
@@ -1038,4 +1038,54 @@ final class AnimalKingdomIRCreationTests: XCTestCase {
     expect(selectionSet.parentType).to(equal(GraphQLObjectType.mock("Height")))
     expect(actual).to(shallowlyMatch(self.expected))
   }
+
+  // MARK: - Referenced Fragment Tests
+
+  func test__referencedFragments__AllAnimalsQuery_isCorrect() throws {
+    // given
+    let operation = compilationResult.operations.first { $0.name == "AllAnimalsQuery" }
+
+    // when
+    let expected: [CompilationResult.FragmentDefinition] = [
+      "HeightInMeters",
+      "WarmBloodedDetails",
+      "PetDetails"
+    ].map { expectedName in
+      compilationResult.fragments.first(where:{ $0.name == expectedName })!
```

Comment:
This check looks like it's enforcing that the expected fragment name must be found but should the test also be validating that there are only the three referenced fragments?